lib/scripts.js

```diff
@@ -314,9 +314,31 @@ var scripts = {
         resolve(null);
       });
     }
-    var redlock = new Redlock(queue.clients, queue.redlock);
-    return redlock.lock(job.lockKey(), queue.LOCK_RENEW_TIME)
-    .then(job.queue.client.hset(job.queue.toKey(job.jobId), "lockAcquired", 1));
+
+    var redlock;
+    if (ensureActive) {
+      var isJobInList = this._isJobInList.replace(/KEYS\[1\]/, '"' + job.queue.toKey('active') + '"')
+        .replace('return item', 'if item')
+        .replace(/ARGV\[1\]\)/, '"' + job.jobId + '")');
+      var lockAcquired = 'and redis.call("HSET", "' + queue.toKey(job.jobId) + '", "lockAcquired", "1")';
+      var success = 'then return 1 else return 0 end';
+      var opts = {
+        lockScript: function(lockScript) {
```

Comment:
ok, so here we can actually "monkey patch" the redlock code. I was actually wondering how we could solve the issue with ensuring that the job is in the active list while getting a redlock atomically. But how robust is this when new versions of redlock are released? Isn't there a risk that our code would stop working?